### PR TITLE
fix(kuma-cp): use stable hashes for outbound cluster names

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by lahabana — COMMENTED** [review]
The change makes sense to me. Let's make sure the orders are maintained too so that we avoid accidental snapshot invalidation.
I'm wondering if this might solve some of the flakes we've seen in the past

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/plugins/policies/core/xds/meshroute/listeners.go [hunk 1]
```diff
@@ -118,7 +77,18 @@ func makeSplit(
...
+		clusterName, _ := tags.Tags(ref.Tags).
+			WithTags(mesh_proto.ServiceTag, ref.Name).
+			DestinationClusterName(nil)
+
+		// The mesh tag is present here if this destination is generated
+		// from a cross-mesh MeshGateway listener virtual outbound.
+		// It is not part of the service tags.
+		if mesh, ok := ref.Tags[mesh_proto.MeshTag]; ok {
+			// The name should be distinct to the service & mesh combination
+			clusterName = fmt.Sprintf("%s_%s", clusterName, mesh)
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/plugins/policies/core/xds/meshroute/listeners.go:89]: Probably not. But more of an existing cross mesh issue

[the remark this answers is not in the document]
Comment (lobkovilya) [reply pkg/plugins/policies/core/xds/meshroute/listeners.go:89]: do we need/have an issue to track that?

**Review by lobkovilya — APPROVED** [review]

File: pkg/plugins/policies/core/xds/meshroute/listeners.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/plugins/policies/core/xds/meshroute/listeners.go:89]: https://github.com/kumahq/kuma/issues/8108

**Pull request merged**